### PR TITLE
Zero value Id in relations (Issue #223)

### PR DESCRIPTION
**Pull request by Pencroff**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -658,7 +658,9 @@ export default {
         } else if (def.type === 'hasOne') {
           if (localKey) {
             prop.get = function () {
-              return get(this, localKey) ? definition.getResource(relationName).get(get(this, localKey)) : undefined
+              let key = get(this, localKey);
```
Comment (jmdobry) [inline src/utils.js]: The value of `key` here is either going to be a string, a number, `null` or `undefined`.

File: src/utils.js [hunk 2]
```diff
@@ -658,7 +658,9 @@ export default {
...
+              let hasKey = isEmpty(key);
```
Comment (jmdobry) [inline src/utils.js]: `isEmpty` returns `true` for `0`, is that what you wanted?

File: karma.start.js [hunk 3]
```diff
@@ -4,6 +4,8 @@ var store, DSUtils, dsHttpAdapter, dsLocalStorageAdapter, p1, p2, p3, p4, p5, p6
 var Post, User, Organization, Comment, Profile;
 var user1, organization2, comment3, profile4;
 var comment11, comment12, comment13, organization14, profile15, user10, user16, user17, user18, organization15, user19, user20, comment19, user22, profile21;
+var Item, Client;
```
Comment (jmdobry) [inline karma.start.js]: Just declare your Item and Client resources and your test data right in the test itself, rather than modifying `karma.start.js`

**Comment by jmdobry** [comment]
@Pencroff Please remove your changes to `dist/` files from your PR, per [CONTRIBUTING.md](https://github.com/js-data/js-data/blob/master/CONTRIBUTING.md), thanks.

File: src/utils.js (hunk 2 above)
Comment (jmdobry) [inline src/utils.js]: So basically, this line should instead be `let hasKey = (typeof key === 'number' && key >= 0) || (typeof key === 'string' && key)`

File: src/utils.js (hunk 1 above)
Comment (jmdobry) [inline src/utils.js]: Also, did you run the full build? Because the build disallows semi-colons.

File: src/datastore/async_methods/loadRelations.js [hunk 4]
```diff
@@ -54,6 +54,9 @@ module.exports = function loadRelations (resourceName, instance, relations, opti
             }
           }
 
+          let defKey = def.localKey ? DSUtils.get(instance, def.localKey) : null
```
Comment (jmdobry) [inline src/datastore/async_methods/loadRelations.js]: If you implement my suggested change below, then your changes to `loadRelations.js` are unnecessary, and can be removed.

File: src/utils.js (hunk 2 above)
Comment (Pencroff) [inline src/utils.js]: What do you think about more simple expression?
`let hasDefKey = !!(defKey || defKey === 0)`

File: karma.start.js (hunk 3 above)
Comment (Pencroff) [inline karma.start.js]: I will moved to `loadRelations.test.js`

**Comment by Pencroff** [comment]
I was confusing with commits. But for now it looks fine. Take a look on my last changes.

**Comment by jmdobry** [comment]
Your changes to anything in the `dist/` should not be committed.

File: src/utils.js [hunk 5]
```diff
@@ -658,7 +658,9 @@ export default {
         } else if (def.type === 'hasOne') {
           if (localKey) {
             prop.get = function () {
-              return get(this, localKey) ? definition.getResource(relationName).get(get(this, localKey)) : undefined
+              let key = get(this, localKey)
+              let hasKey = !!(key || key === 0)
```
Comment (jmdobry) [inline src/utils.js]: This works

File: test/both/mocha.start.js [hunk 6]
```diff
@@ -0,0 +1,371 @@
+/*global assert:true */
```
Comment (jmdobry) [inline test/both/mocha.start.js]: Why did you add this new file?

File: dist/js-data.js [hunk 7]
```diff
@@ -1264,7 +1264,9 @@ return /******/ (function(modules) { // webpackBootstrap
 	        } else if (def.type === 'hasOne') {
```
Comment (jmdobry) [inline dist/js-data.js]: You shouldn't commit any changes to this file.

File: dist/js-data-debug.js [hunk 8]
```diff
@@ -1269,7 +1269,9 @@ return /******/ (function(modules) { // webpackBootstrap
 	        } else if (def.type === 'hasOne') {
```
Comment (jmdobry) [inline dist/js-data-debug.js]: You shouldn't commit any changes to this file.

File: karma.start.js [hunk 9]
```diff
@@ -349,6 +349,8 @@ beforeEach(function () {
     user: user22
```
Comment (jmdobry) [inline karma.start.js]: You shouldn't commit any changes to this file.

File: src/datastore/async_methods/loadRelations.js (hunk 4 above)
Comment (jmdobry) [inline src/datastore/async_methods/loadRelations.js]: You should not longer have any changes to this file.

**Comment by jmdobry** [comment]
Can you please squash your changes into 1 commit and then update your branch with master?

**Comment by Pencroff** [comment]
Close in case of new one - #237
